### PR TITLE
Fix thread safety of softmax prediction.

### PR DESCRIPTION
**Pull request by trivialfis**
Make the tranform function const.

See failing CI: https://xgboost-ci.net/blue/organizations/jenkins/xgboost/detail/PR-7092/20/pipeline .

### CODE REVIEW

**Review by hcho3 — APPROVED** [review]

**Comment by trivialfis** [comment]
@hcho3 Could you please help take a look into freezing libomp on travis when you are available?
